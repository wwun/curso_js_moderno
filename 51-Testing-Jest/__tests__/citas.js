import Citas from '../js/classes/Citas';

describe('probar clase Citas', () => {

    const citas = new Citas();

    const id = Date.now();

    test('agregar una nueva cita', () => {
        const citaObj = {
            id,
            mascota: 'Hook',
            propietario: 'William',
            telefono: '7788583949',
            fecha: '10-12-2023',
            hora:'10:30',
            sintomas: 'come'
        };

        citas.agregarCita(citaObj);

        expect(citas).toMatchSnapshot();
    });

    test('actualizar cita', () => {
        const citaActualizada = {
            id,
            mascota: 'Nuevo nombre',
            propietario: 'William',
            telefono: '7788583949',
            fecha: '10-12-2023',
            hora:'10:30',
            sintomas: 'come'
        };
        citas.editarCita(citaActualizada);

        expect(citas).toMatchSnapshot();
    });

    test('eliminar cita', () => {
        citas.eliminarCita(id);
        expect(citas).toMatchSnapshot();
    })
});