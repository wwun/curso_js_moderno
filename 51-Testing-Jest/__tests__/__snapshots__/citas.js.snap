// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`probar clase Citas actualizar cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-12-2023",
      "hora": "10:30",
      "id": 1725572916391,
      "mascota": "Nuevo nombre",
      "propietario": "William",
      "sintomas": "come",
      "telefono": "7788583949",
    },
  ],
}
`;

exports[`probar clase Citas agregar una nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-12-2023",
      "hora": "10:30",
      "id": 1725572916391,
      "mascota": "Hook",
      "propietario": "William",
      "sintomas": "come",
      "telefono": "7788583949",
    },
  ],
}
`;

exports[`probar clase Citas eliminar cita 1`] = `
Citas {
  "citas": [],
}
`;
